packages/aws-cdk/lib/cli.ts

```diff
@@ -547,16 +548,19 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
             if (args.changeSetName) {
               throw new Error('--change-set-name cannot be used with method=direct');
             }
+            if (args.importExistingResources !== undefined) {
+              throw new Error('--import-existing-resources cannot be used with method=direct');
+            }
```

Comment:
possible to add a unit for this ?

Reply:
Afaik there's no unit test for cli.ts, so I added a test case in cli-integ.

Reply:
But thanks anyway. I forgot that `args.importExistingResources` is  by default `false` not `undefined`, so there was a bug here...